### DISCUSSION TITLE
HEAD is now at XXX Merge YYY into ZZZ

### DISCUSSION THREAD

**Discussion by lzap** [actions/checkout#885] [Q&A] [answered]
Hey,

we were investigating some weird compiler/linter errors the other day and we could not reproduce locally. It was slightly older branch and I noticed this message at the very end of the checkout action:

```
HEAD is now at 80be8bd Merge 561625b58f0f9c5268d5cd671a423d94ff17ceeb into a729d518ef462b0088d176a1b5836f259a1985de
```

Does this mean that checkout action does perform some kind of rebase or pre-merge? Is there any setting to turn this off? I would like to be on the branch HEAD (561625b) and not on a merge commit (80be8bd) because some compilers or linters (e.g. go import) are very picky about import order or unused keywords and this is really hard to troubleshoot.

https://github.com/RHEnVision/provisioning-backend/runs/7763538534?check_suite_focus=true

Thanks!

**Comment by nonameolsson** [comment, marked as answer]
Is this what you are looking for? https://github.com/actions/checkout#checkout-pull-request-head-commit-instead-of-merge-commit

**Reply by lzap** [reply]
Yes thank you I figured it out and forgot to mark this one as solved.
